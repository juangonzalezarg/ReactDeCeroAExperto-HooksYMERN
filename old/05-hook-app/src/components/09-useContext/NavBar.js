import React from 'react';
import {Link, NavLink} from 'react-router-dom';

export const NavBar = () => {
  return (
    <nav className="navbar navbar-expand-sm navbar-dark bg-dark">
        <div className="container-fluid">
            
            <Link to="/" className="navbar-brand">useContext</Link>

            <div className="collapse navbar-collapse" id="navbarNavAltMarkup">
                <div className="navbar-nav">
                    <NavLink exact = "true" activeclassname="active"  to="/" className="nav-link">Home</NavLink>
                    <NavLink exact = "true" activeclassname="active" to="/about" className="nav-link">About</NavLink>                        
                    <NavLink exact = "true" activeclassname="active" to="/login" className="nav-link">Login</NavLink>                                                
                </div>
            </div>
        </div>
    </nav>
  )
}
