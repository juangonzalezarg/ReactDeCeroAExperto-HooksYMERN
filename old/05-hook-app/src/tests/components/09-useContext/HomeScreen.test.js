import { mount } from 'enzyme';
import React from 'react';
import { HomeScreen } from '../../../components/09-useContext/HomeScreen';
import { UserContext } from '../../../components/09-useContext/UserContext';

describe('Prueba en <HomeScreen />', () => { 
    
    const user = {
        name: 'Juan',
        email: 'juan@gmail.com'
    }

    const wrapper = mount(
        <UserContext.Provider value={{
            user
        }}>
            <HomeScreen/>
        </UserContext.Provider>
    );

    test('debe de mostrarse correctamente', () => { 
        
        expect(wrapper).toMatchSnapshot();

     });

 });

