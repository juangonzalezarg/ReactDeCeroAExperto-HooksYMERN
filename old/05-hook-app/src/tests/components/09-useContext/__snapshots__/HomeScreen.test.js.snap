// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <HomeScreen /> debe de mostrarse correctamente 1`] = `
<div>
  <h1>
    HomeScreen
  </h1>
  <hr />
  <pre>
    {
   "name": "Juan",
   "email": "juan@gmail.com"
}
  </pre>
</div>
`;
